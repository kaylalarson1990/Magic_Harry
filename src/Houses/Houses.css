.houseCard {
    align-items: center;
    background-color: white;
    border: .5px solid #EBECEC;
    border-radius: 3px;
    box-shadow: black;
    display: flex;
    flex-direction: column;
    font-family: 'Open Sans';
    height: 200px;
    margin: 15px;
    width: 300px;
}

h2 {
    text-align: center;
    font-weight: 600;
}

.favorite-btn {
    margin-top: 10px;
    width: 35px;
}