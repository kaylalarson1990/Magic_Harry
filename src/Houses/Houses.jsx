import React, { Component } from "react";
import PropTypes from "prop-types";
import ReactCardFlip from "react-card-flip";
import "./Houses.css";

export class Houses extends Component {
  constructor() {
    super();
    this.state = {
      isFlipped: false
    };
  }

  handleClick = e => {
    e.preventDefault();
    this.setState(prevState => ({ isFlipped: !prevState.isFlipped }));
  };

  render() {
    const {
      name,
      mascot,
      id,
      headOfHouse,
      houseGhost,
      founder,
      favoriteCard,
      info
    } = this.props;
    return (
      <ReactCardFlip isFlipped={this.state.isFlipped} flipDirection="vertical">
        <article className="houseCard" key="front">
          <h2>{name}</h2>
          <button onClick={this.handleClick}>
            Click here for more details
          </button>
          <button onClick={e => favoriteCard(info.id)} className={`favorite-btn ${info.favorite && "favorite"}`}>
              Favorite
          </button>
        </article>
        <article className="houseCard" key="back">
          <p>
            <span>Mascot:</span> {mascot}
          </p>
          <p>
            <span>Head Of House:</span> {headOfHouse}
          </p>
          <p>
            <span>House Ghost:</span> {houseGhost}
          </p>
          <p>
            <span>Founder:</span> {founder}
          </p>
          <button onClick={this.handleClick}>Return to Front</button>
        </article>
      </ReactCardFlip>
    );
  }
}

Houses.propTypes = {
  name: PropTypes.string,
  mascot: PropTypes.string,
  house: PropTypes.string,
  school: PropTypes.string,
  id: PropTypes.number,
  headOfHouse: PropTypes.string,
  houseGhost: PropTypes.string,
  founder: PropTypes.string
};

export default Houses;
