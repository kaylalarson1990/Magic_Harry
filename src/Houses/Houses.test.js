import React from "react";
import { shallow } from "enzyme";
import Houses from "./Houses";
//... 23,24,25,29,30

describe("Houses", () => {
  let wrapper, mockFunc, instance;
  let event = {
    preventDefault: jest.fn()
  };

  let props = {
    name: "Kayla",
    mascot: "Bengal",
    house: "Apartment",
    school: "Turing",
    id: 1,
    headOfHouse: "Sergio",
    houseGhost: "Klio",
    founder: "Sergio"
  };

  beforeEach(() => {
    mockFunc = jest.fn();
    wrapper = shallow(<Houses {...props} />);
    instance = wrapper.instance();
  });

  it("should match snapshot", () => {
    wrapper.setState({ error: "" });
    expect(wrapper).toMatchSnapshot();
  });

  it("should be able to toggle favorite", () => {
    instance.addFavorite = mockFunc;
    wrapper.find(".favorite-btn").simulate("click");
    expect(instance.addFavorite).not.toHaveBeenCalled();
  });

  it("should set flipped to state", () => {
    expect(wrapper.state("isFlipped")).toEqual(false);
    instance.handleClick(event);
    expect(wrapper.state("isFlipped")).toEqual(true);
  });

  it("should set favorite to state", () => {
    expect(wrapper.state("favorited")).toEqual(false);
    instance.addFavorite();
    expect(wrapper.state("favorited")).toEqual(true);
  });
});
