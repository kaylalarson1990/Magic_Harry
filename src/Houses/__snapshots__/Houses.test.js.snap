// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Houses should match snapshot 1`] = `
<t
  cardStyles={
    Object {
      "back": Object {},
      "front": Object {},
    }
  }
  cardZIndex="auto"
  containerStyle={Object {}}
  flipDirection="vertical"
  flipSpeedBackToFront={0.6}
  flipSpeedFrontToBack={0.6}
  infinite={false}
  isFlipped={false}
>
  <article
    className="houseCard"
    key="front"
  >
    <h2>
      Kayla
    </h2>
    <button
      onClick={[Function]}
    >
      Click here for more details
    </button>
    <img
      alt="favorite icon"
      className="favorite-btn"
      onClick={[Function]}
      src="inactive.png"
    />
  </article>
  <article
    className="houseCard"
    key="back"
  >
    <p>
      <span>
        Mascot:
      </span>
       
      Bengal
    </p>
    <p>
      <span>
        Head Of House:
      </span>
       
      Sergio
    </p>
    <p>
      <span>
        House Ghost:
      </span>
       
      Klio
    </p>
    <p>
      <span>
        Founder:
      </span>
       
      Sergio
    </p>
    <button
      onClick={[Function]}
    >
      Return to Front
    </button>
  </article>
</t>
`;
